### PR TITLE
USHIFT-4743: Add user docs for gateway api

### PR DESCRIPTION
**Pull request by pacevedom**
**Which issue(s) this PR addresses**:

Closes #<Issue Number>

### CODE REVIEW

File: docs/user/howto_gateway_api.md [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+    status: "True"
+    type: Accepted
+
+# Service mesh operator and istio controller
+$ oc get deployment -n openshift-gateway-api
+NAME                           READY   UP-TO-DATE   AVAILABLE   AGE
+istiod-openshift-gateway-api   1/1     1            1           26h
+servicemesh-operator3          1/1     1            1           26h
+```
+These resources are handled by MicroShift and will always be present.
```
Comment (ggiguash) [inline docs/user/howto_gateway_api.md:83]: It's unclear. Can we rephrase / add more info?

File: docs/user/howto_gateway_api.md [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+
+## Configuring Gateways
+In order to use the feature we only need an application that we want to expose through a gateway, a gateway, and some routes.
```
Comment (ggiguash) [inline docs/user/howto_gateway_api.md:86]:
```suggestion
In order to use the feature we need an application that we want to expose through a gateway, a gateway, and some routes.
```

File: docs/user/howto_gateway_api.md [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+
+First we create a sample application:
+```bash
+$ oc create -f microshift/test/assets/hello-microshift.yaml
```
Comment (ggiguash) [inline docs/user/howto_gateway_api.md:90]: We usually assume we're inside `microshift` repo.

File: docs/user/howto_gateway_api.md [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+## Configuring Gateways
+In order to use the feature we only need an application that we want to expose through a gateway, a gateway, and some routes.
+
+First we create a sample application:
+```bash
+$ oc create -f microshift/test/assets/hello-microshift.yaml 
+pod/hello-microshift created
+
+$ oc create -f microshift/test/assets/hello-microshift-service.yaml 
+service/hello-microshift created
```
Comment (ggiguash) [inline docs/user/howto_gateway_api.md:94]: I'm not sure we need outputs for the creation commands. Let's just give the commands and check the pod/svc output?

File: docs/user/howto_gateway_api.md [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+    port: 8080
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: All
+
+$ oc create -f gatewayapi/gateway.yaml 
+gateway.gateway.networking.k8s.io/demo-gateway created
+
+$ oc get gateway -n openshift-gateway-api demo-gateway -o yaml | yq '.status'
```
Comment (ggiguash) [inline docs/user/howto_gateway_api.md:126]: Can we further filter out the output to make it shorter? Probably the address part is the most interesting one, right?

File: docs/user/howto_gateway_api.md [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+HTTP/1.1 404 Not Found
+date: Wed, 06 Nov 2024 16:17:07 GMT
+server: istio-envoy
+content-length: 0
+```
+
+## Uninstall
+Gateway API is installed through RPM packages, therefore removing them will erase the files for all the new manifests.
+```bash
+$ sudo dnf remove -y microshift-gateway-api-release-info microshift-gateway-api
```
Comment (ggiguash) [inline docs/user/howto_gateway_api.md:330]: Let's remove the output of the command?

File: docs/user/howto_gateway_api.md [hunk 7]
```diff
@@ -0,0 +1,411 @@
...
+  clusterrole.rbac.authorization.k8s.io "istio-reader-clusterrole-openshift-gateway-api-openshift-gateway-api" deleted
+  clusterrole.rbac.authorization.k8s.io "istiod-clusterrole-openshift-gateway-api-openshift-gateway-api" deleted
+  clusterrole.rbac.authorization.k8s.io "istiod-gateway-controller-openshift-gateway-api-openshift-gateway-api" deleted
+  clusterrole.rbac.authorization.k8s.io "servicemesh-operator3" deleted
+  
+  $ oc delete clusterrolebindings.rbac.authorization.k8s.io istio-reader-clusterrole-openshift-gateway-api-openshift-gateway-api istiod-clusterrole-openshift-gateway-api-openshift-gateway-api istiod-gateway-controller-openshift-gateway-api-openshift-gateway-api servicemesh-operator3
+  clusterrolebinding.rbac.authorization.k8s.io "istio-reader-clusterrole-openshift-gateway-api-openshift-gateway-api" deleted
+  clusterrolebinding.rbac.authorization.k8s.io "istiod-clusterrole-openshift-gateway-api-openshift-gateway-api" deleted
+  clusterrolebinding.rbac.authorization.k8s.io "istiod-gateway-controller-openshift-gateway-api-openshift-gateway-api" deleted
+  clusterrolebinding.rbac.authorization.k8s.io "servicemesh-operator3" deleted
```
Comment (ggiguash) [inline docs/user/howto_gateway_api.md:383]: Can we query / grep the resources like for CRDs?

File: docs/user/howto_gateway_api.md [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+  ```
+* Remove CRD definitions.
+  ```bash
+  $ oc get crd | grep -E "gateway|istio" | awk '{print $1}' | xargs oc delete crd
```
Comment (ggiguash) [inline docs/user/howto_gateway_api.md:387]: Let's not include the output?

File: docs/user/howto_gateway_api.md (hunk 1 above)
Comment (pacevedom) [reply docs/user/howto_gateway_api.md]: Removed it, as its implicit by installing the rpm package.

File: docs/user/howto_gateway_api.md [hunk 9]
```diff
@@ -0,0 +1,308 @@
...
+* Remove `openshift-gateway-api` namespace. This will remove all the namespaced resources.
+  ```bash
+  $ oc delete namespace openshift-gateway-api 
+  namespace "openshift-gateway-api" deleted
+  ```
+* Remove `ClusterRole` and `ClusterRoleBindings`.
+  ```bash
+  $ oc get clusterrole | grep -E "openshift-gateway-api|servicemesh-operator" | awk '{print $1}' | xargs oc delete clusterrole
+  
+  $ oc delete clusterrolebinding | grep -E "openshift-gateway-api|servicemesh-operator" | awk '' | xargs oc delete clusterrolebinding
```
Comment (ggiguash) [inline docs/user/howto_gateway_api.md:303]: Looks like awk command is not valid.

**Pull request merged**